x/programs/runtime/runtime.go

```diff
@@ -189,6 +189,8 @@ func (r *WasmRuntime) Call(_ context.Context, name string, params ...uint64) ([]
 	case int64:
 		value := uint64(result.(int64))
 		return []uint64{value}, nil
+	case nil:
+		return []uint64{}, nil
```

Comment:
better to return `nil` or empty `[]uint64` here?

Reply:
This should return nil as it appears to be a valid resposne. Can you also add a note on the interface :pray: ?

since we are here we can also clean up this switch a bit.

```go
switch v := result.(type) {
case int32, int64:
    value := uint64(v)
    return []uint64{value}, nil
case nil:
    return nil, nil
default:
    return nil, fmt.Errorf("invalid result type: %T", result)
}
```
https://github.com/ava-labs/hypersdk/blob/f735d9422ef7db029ec64bb26a2bf9af563dac98/x/programs/runtime/dependencies.go#L32-L33

Reply:
Also lets add some docs on the nil case Ideally we could reference the actual spec.

Reply:
tried to do this in go, but we need a type assertion in both cases
```go
value := int64(v.(int32))
```
or in the int64 case
```go
value := v.(int64)
```
Is there a workaround for this?

Reply:
actually there isn't, so we can skip this.